### PR TITLE
Feature/add designate

### PR DESCRIPTION
**Pull request by gregoiremahe**
- Add Designate OpenStack component
- Update ansible syntax to avoid deprecated warnings
- Improve delay time to play a playbook (kind of cache on dynamic hosts script)
- Transforms many playbooks into many roles with only one playbook
- Update documentation

### CODE REVIEW

File: ansible/dynhosts [hunk 1]
```diff
@@ -188,6 +188,15 @@ def main():
         elif args.host:
             output = to_json(inventory.get_host(args.host))
         print(output)
+
+        # Write inventory file in /etc/ansible/hosts
+        with open("/etc/ansible/hosts", "w+") as hostfile:
+            hostfile.write('#!/usr/bin/env python\n\n')
```
Comment (arnaudmorin) [inline ansible/dynhosts:194]: cant you write direct json / yaml file?
https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html#hosts-and-groups

Comment (gregoiremahe) [reply ansible/dynhosts:194]: Just checked out, we actually can specify a json file as extra vars. doing..

Comment (gregoiremahe) [reply ansible/dynhosts:194]: done

**Comment by gregoiremahe** [comment]
Done reverting last commit to generate a hosts file with hardcoded json extra vars

**Review by arnaudmorin — APPROVED** [review]

**Pull request merged**